src/kibana/components/vislib/visualizations/time_marker.js

```diff
@@ -0,0 +1,41 @@
+define(function () {
+  return function TimeMarkerFactory(d3) {
+    function TimeMarker(xScale, height) {
+      if (!(this instanceof TimeMarker)) {
+        return new TimeMarker(xScale, height);
+      }
+
+      this.xScale = xScale;
+      this.height = height;
+      this.lineClass = 'time-marker';
+      this.time = new Date().getTime();
```

Comment:
Can you add an option to pass this in, preferably having it take an array so we can pass in multiple?